integration-cli/docker_cli_run_unix_test.go

```diff
@@ -315,3 +315,17 @@ func (s *DockerSuite) TestRunWithSwappinessInvalid(c *check.C) {
 		c.Fatalf("failed. test was able to set invalid value, output: %q", out)
 	}
 }
+
+func (s *DockerSuite) TestRunWithMemoryReservation(c *check.C) {
+	testRequires(c, memoryReservationSupport)
+	dockerCmd(c, "run", "--memory-reservation", "200M", "busybox", "true")
+}
+
+func (s *DockerSuite) TestRunWithMemoryReservationInvalid(c *check.C) {
+	testRequires(c, memoryLimitSupport)
+	testRequires(c, memoryReservationSupport)
+	out, _, err := dockerCmdWithError("run", "-m", "500M", "--memory-reservation", "800M", "busybox", "true")
+	if err == nil {
```

Comment:
this err nilness check does not guarantee the error thrown was about memory reservation. It could still fail about an invalid flag for instance and still this check would be false and test passes w/o having checked anything, we should at least check for command output containing something about memory reservation error. You could confirm this by just introducing a typo in --memory-reservation flag and see that this test passes. We need something like:

```
out, _, err := dockerCmdWithError("run", "--memory-reservation", "800M", "busybox", "true")
c.Assert(err, check.NotNil)
expected := "Invalid value 800M for memory reservation."
if !strings.Contains(strings.TrimSpace(out), expected) {
    // fail
}
```